app/controllers/api/migrations.php

```diff
@@ -0,0 +1,824 @@
+<?php
+
+use Appwrite\Auth\OAuth2\Firebase as OAuth2Firebase;
+use Appwrite\Event\Delete;
+use Appwrite\Event\Event;
+use Appwrite\Event\Migration;
+use Appwrite\Extend\Exception;
+use Utopia\Database\Helpers\ID;
+use Appwrite\Utopia\Database\Validator\Queries\Migrations;
+use Appwrite\Utopia\Request;
+use Appwrite\Utopia\Response;
+use Utopia\App;
+use Utopia\Database\Database;
+use Utopia\Database\DateTime;
+use Utopia\Database\Document;
+use Utopia\Database\Query;
+use Utopia\Database\Validator\UID;
+use Utopia\Transfer\Sources\Appwrite;
+use Utopia\Transfer\Sources\Firebase;
+use Utopia\Transfer\Sources\NHost;
+use Utopia\Transfer\Sources\Supabase;
+use Utopia\Validator\ArrayList;
+use Utopia\Validator\Host;
+use Utopia\Validator\Integer;
+use Utopia\Validator\Text;
+use Utopia\Validator\URL;
+use Utopia\Validator\WhiteList;
+
+include_once __DIR__ . '/../shared/api.php';
+
+App::get('/v1/migrations')
```

Comment:
Let's reorder the routes so that we have them ordered like create, list, read, update, delete (post, get, get, patch, put, delete) for each resource type. Take a look at the other controllers for an example